lib/Dialect/FIRRTL/Transforms/LowerLayers.cpp

```diff
@@ -155,33 +168,37 @@ class LowerLayersPass : public LowerLayersBase<LowerLayersPass> {
   /// Indicates exclusive access to modify the circuitNamespace and the circuit.
   llvm::sys::SmartMutex<true> *circuitMutex;
 
+  SymbolTable *symbolTable;
+
   /// A map of layer blocks to module name that should be created for it.
   DenseMap<LayerBlockOp, StringRef> moduleNames;
 
-  /// The design-under-test (DUT) as indicated by the presence of a
-  /// "sifive.enterprise.firrtl.MarkDUTAnnotation".  This will be null if no
-  /// annotation is present.
-  FModuleOp dut;
+  DenseMap<SymbolRefAttr, hw::OutputFileAttr> layerOutputDirTable;
 
-  /// The directory for verification collateral as indicated by the presence of
-  /// a "sifive.enterprise.firrtl.TestBenchDirAnnotation".  Empty if not
-  /// specified.
-  StringRef testBenchDir;
+  hw::OutputFileAttr getLayerOutputDir(SymbolRefAttr ref) {
+    return layerOutputDirTable[ref];
+  }
```

Comment:
This is a private member function wrapping access to a private member. This may be cleaner as just the latter.